[KNOXPOD] how to connect to data templates

Aylr opened the pull request:
Changes proposed in this pull request:
- This PR is another and another and another revision on how to connect to data guides.

Previous Design Review notes:
- This addresses feedback in the knoxpod design review on 5/17 and 5/21

GE-149

Aylr committed:
* removed didactic content from postgres
* simplified wording in postgres article
* new stub articles
* linting

Commit:
Update next_steps.md

Commit:
Update next_steps.md

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/database/postgres.md:
````diff
@@ -1,152 +1,111 @@
+```   
 
-### 3. Load the DataContext into memory
+### 3. Load your project's DataContext into memory
````
<3

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/database/postgres.md:
````diff
@@ -1,152 +1,111 @@
 
-```python file=../../../../integration/code/connecting_to_your_data/database/postgres.py#L1-L67
-```
+To view the full script [see it on GitHub](https://github.com/great-expectations/great_expectations/blob/knoxpod/integration/code/connecting_to_your_data/database/postgres.py)
````
Love this as well

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/database/postgres.md:
```diff
@@ -1,152 +1,111 @@
-  - Add test for yaml here
+**Ideas**
+- Use the convenience function the `datasource new` notebooks use?
+- Port said function to the data context itself? (probably!)
```
Yes and yes, if possible

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/database/postgres.md:
````diff
@@ -1,152 +1,111 @@
 ```
 
-### 2. Determine how to add credentials to configuration
+### 2. Choose a credential handling method
````
Nit-picky, but this title sounds a bit .. clunky for me.

Aylr replied:
Agreed!

Shinnnyshinshin reviewed (commented):
Current impression for the `postgres` template is that I like it overall :) Even with most of the didactic content taken out, I still felt that the guide flowed nicely. Seeing it "in person" really drove that point home for me.  I liked the changes to the titles (other than the one nit-pick) and the link to full script on Github at the end. 

Regarding how we should handle the CLI workflow (for pandas and actually postgres is involved too), I've added an update to the JIRA task (GE-149).  My current feeling is that I'd like to push for parity between the CLI + all the datasource combinations by having  both the `InferredAssetDataConnector` and `RuntimeDataConnector` by default. We can talk more about this. 

PS. I also tried fixing a broken link in the docs that was making the build fail, but was only partially successful :/

Aylr committed:
* renamed pandas script to avoid silly import error to improve dev loop
* harmonized pandas article with postgres article

Aylr committed:
missed a file

Aylr committed:
corrected line errors after running linter

Aylr committed:
core skills emphasized in next steps and articles

Aylr committed:
oops

Aylr committed:
* tighter next steps emphasizing core skills
* removed article stubs

Aylr committed:
Addressed feedback - tighter titles

Aylr committed:
remove obsolete files

Aylr committed:
more cleanup

Aylr committed:
* where to run code idea

Aylr committed:
* reusable Prerequisites component

Aylr committed:
* tighter wording
* more cleanup
* removed jsx errors

Aylr committed:
oops

Aylr committed:
* howto templates now have modular where to run code block
* howto templates now have tabs for python/yaml configuration/adding
* additional scripts for postgres and pandas

Aylr committed:
oops forgot key new files

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/components/where_to_run_code.md:
```diff
@@ -0,0 +1,37 @@
+
+<Tabs
+  defaultValue='cli'
+  values={[
```
```suggestion
<Tabs
  groupId="yaml-or-python"
  defaultValue='cli'
  values={[
```

Shinnnyshinshin replied:
This way the `YAML` , and `python` section will stay the same for all tabs.  I think CLI can be its own thing with one additional sentence that I described in a separate comment

Shinnnyshinshin commented on integration/code/connecting_to_your_data/filesystem/pandas_python_example.py:
```diff
@@ -0,0 +1,47 @@
+        }
+    },
+}
+
```
Will need to add the `InferredAssetDataConnector` here and in the `pandas_example.py` file too. What to do with the `base_directory` is an open question, unfortunately.  Can we discuss how to approach adding / testing the `InferredAssetDataConnector` for Pandas/Spark before merging this PR?

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/components/where_to_run_code.md:
````diff
@@ -0,0 +1,37 @@
+```console
+great_expectations --v3-api datasource new
+```
+
````
Since `CLI` is its own tab here, I think one more sentence saying something like : "running the CLI command will open up a separate Notebook, and you can follow along in the YAML workflow below" would help users get oriented. Also with the added bonus that `YAML` is the default tab that is selected, the guide will read seamlessly :)

Commit:
[KNOXPOD] CLI Datasource Notebooks support multiple DataConnectors (#2814)

- Added credentials to SqlAlchemyExecutionEngine schema, and corresponding tests.

Aylr committed:
* pandas filesystem article + yaml script updated and working
* where to run code CLI messaging improved

Aylr committed:
Merge branch 'howtow-template-GE-149' of github.com:great-expectations/great_expectations into howtow-template-GE-149

Aylr committed:
* pandas filesystem python corrected and working

Aylr committed:
* simpler fixture for integration tests for howto connect guides

Aylr committed:
* postgres article finished
* postgres scripts adjusted

Aylr committed:
fixed error in con string

Aylr commented on integration/fixtures/no_datasources/great_expectations/great_expectations.yml:
```diff
@@ -13,19 +13,8 @@ config_version: 3.0
-          - default_identifier_name
-        module_name: great_expectations.datasource.data_connector
-        class_name: RuntimeDataConnector
+datasources: {}
+
```
All our how to connect guides should start with no datasources, hence the fixture change and rename.

Shinnnyshinshin replied:
<3

Aylr committed:
tabs linked

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/components/where_to_run_code.md:
```diff
@@ -0,0 +1,38 @@
+  ]}>
+  <TabItem value="cli">
+
+If you use the Great Expectations CLI, run this command to automatically generate a pre-configured jupyter notebook. Then you can follow along in the YAML-based workflow below:
```
I think this will address any potential "mismatch" we have between the tabs here and the tabs below. Thanks!

Aylr committed:
removed todo

Shinnnyshinshin commented on docs/guides/connecting_to_your_data/filesystem/pandas.md:
````diff
@@ -1,136 +1,131 @@
-
-Congratulations! If no errors are shown here, you've just connected to your data on a filesystem using pandas!
+Add the path to your CSV in the `path` key under `runtime_parameters`.
 
-:::tip Next steps
-After connecting to data, usually people validate data!
-**TODO** insert link or have this be an include.
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_yaml_example.py#L39-L45
+```
+Then load data into the `Validator`.
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_yaml_example.py#L53-L59
+```
 
-Right now a Batch is just Data + Metadata that is recognized by GE. If you want to actually Validate your data with Expectations, you will need a Batch + ExpectationSuite.
-:::
+<Congratulations />
````
Pandas-like how-to-guides will support both a `RuntimeBatchRequest` with path (using a `RuntimeDataConnector`) and a `BatchRequest` with `data_asset_name` (using an `InferredAssetDataConnector`). This will be submitted as a separate PR as discussed during review

Shinnnyshinshin commented on integration/code/connecting_to_your_data/filesystem/pandas_python_example.py:
```diff
@@ -0,0 +1,74 @@
+    runtime_parameters={"path": "<PATH_TO_YOUR_DATA_HERE>"},  # Add your path here.
+    batch_identifiers={"default_identifier_name": "something_something"},
+)
+
```
To add `BatchRequest` here

Shinnnyshinshin reviewed (approved):
LGTM
